073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,139 @@
+These solutions generate certificates signed by arbitrary CAs, requiring the support for more than one root CA certificate.
+In addition these solutions cannot be responsible for rolling updates to components like Kafka, so this should be decoupled from the CA generation.
+This would be highly valuable for organizations with compliance requirements with regard to certificates. 
+
+## Proposal
+
+Strimzi should be updated to decouple the management of the cluster and clients CAs, and the consumption of those CAs by components (such as ZooKeeper, Kafka and cluster operator).
+
+To do this we will introduce new secrets to share the CA certificates and track their lifecycle
+These secrets are different from the ones that are mounted into the components to use a truststore.
+This allows the cluster operator and user operator to manage the truststore secrets independently of how the CAs are being managed and make it easier to reason about which CA certificates are in use.
+These secrets can be updated by multiple processes.
+
+### Secret format
+
+```yaml
+kind: Secret
+metdata:
+  name: ${cluster}-${ca-type}-trust # e.g. foo-cluster-ca-trust
+type: strimzi.io/trust
+data:
+  ${fingerprint}.${state}: <PEM encoded Secret>
+```
+
+The `fingerprint` is the fingerprint/thumbprint of the certificate. I.e. the SHA-1 hash of the ASN.1 DER-encoded form of the X509 certificate. The `state` can be any of the states of the CA certificate trust state machine described below.
+Including the state in the item name facilitates inspection of the Secret.
+
+The existing Secrets, named using `${cluster}-${ca}-certs` and passed via Secret volume mounts to containers would not change under this proposal.
+
+Problems from multiple processes concurrently updating the secret are avoided by using `metadata.resourceVersion` to make conditional requests when updating that Secret. Note also that only one operator makes each state transition.
+
+### CA state machine
+
+The "CA certificate trust state machine" has the following states:
+
+```
+  UNTRUSTED // not yet trusted everywhere it needs to be 
+   |
+   v
+  TRUSTED_UNUSED // trusted everywhere it needs to be, but no EE certificates yet issued
+   |
+   v
+  TRUSTED_USED // trusted everywhere, EE certificates issued
+   |
+   v
+  PHASE_OUT // trusted everywhere, but being phased out
+   |
+   v
+  ZOMBIE // still trusted by some component, but not relied on by any party.
+```
+
+New cluster creation process:
+1. The CaReconciler creates the CA certificates and places them in the relevant Secret with the state UNTRUSTED
+2. The cluster and user operator, during their respective reconcile loops:
+   1. Add the CA certificate to the relevant component truststore secret and change the state to TRUSTED_UNUSED
+   2. Generate end entity certificates using the CA and add the EE certificate to the relevant component keystore secret and change the state to TRUSTED_USED
+
+Update process (e.g. CA key has been updated):
```

Comment:
key is a very overloaded term. Can you qualify this as either public key, private key, or `key` within a `Secret`. (not secret key, which is another thing!)